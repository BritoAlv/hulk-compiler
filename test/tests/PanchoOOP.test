type A(id : string) 
{
    id = id;
    jump() : object => print(self.id);
    greet() : object => print("Call me" @@ self.id);
}

type B(id : string, size : number) inherits A(id)
{
    size = size;
    jump() : object => print("My size is" @@ self.size);
}

type C inherits B { }

let a = new A("Pancho"), b = new B("Jenn", 20), c = new C("John", 30) in 
{
    a.jump();
    b.jump();
    b.greet();
    c.jump();
    c.greet();
};