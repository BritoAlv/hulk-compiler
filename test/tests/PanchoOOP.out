Pancho
My size is 20
Call me Jenn
My size is 30
Call me John

